### ISSUE TITLE
UI Improvements to Date and TimeWidgets

### ISSUE THREAD

**Issue by SaumiaSinghal** [getodk/collect#4024]
As discussed in #3975, to make date and time widgets more consistent with other Collect widgets, @grzesiek2010 proposed to remove `No_date_selected` and `No_time_selected` label, in case the prompt does not have an answer.

In order to avoid having extra empty space between widget buttons (when prompt does not have answer, and when the form has multiple date or time widgets on one screen), We can make use of material components. Material Text Input field can be used to prompt the user to Select date/ Select time, with a leading or trailing icon signifier to display date picker or time picker dialog, as mentioned [here](https://material.io/components/text-fields#anatomy) on Material.io web page.

![89787399-01532100-db3b-11ea-8717-36154a19bc8c](https://user-images.githubusercontent.com/35730054/89945999-dacade00-dc3f-11ea-84e8-ae9614eda95e.png)

The widget after the mentioned UI improvement will look somewhat like this.

Some issues that we need to decide upon if we are going with this UI modification are:
* Whether to use leading or trailing icon specifier
* Which leading /trailing icons to use, I feel it would be best to use the material calendar icon ![baseline_calendar_today_black_18dp](https://user-images.githubusercontent.com/35730054/89944974-47dd7400-dc3e-11ea-9ec9-ea42324d9e0e.png) for date picker dialog, and acess time icon ![baseline_access_time_black_18dp](https://user-images.githubusercontent.com/35730054/89945247-b02c5580-dc3e-11ea-8848-ed340c4ed259.png) for time picker dialog.
* Whether to have a hint text displaying the date/ time format, just to make the widget more appealing 

 Also, I think I have displayed wrongly in the wireframe, that the label text should be `Enter Date`/ `Enter Time`, I think we should stick with `Select Date`/`Select Time` as currently, the user is not allowed to enter date or time in the widgets manually.

**Comment by grzesiek2010** [comment]
It should be a broader discussion because if we decide to refresh those buttons we should review all widgets and implement the same improvements everywhere.
So we probably want to do the same with geo widgets / media widgets etc.
Generally I like this idea but we need also to schedule it somehow. Now we are in the middle of reworking widgets (moving the code to layout files) then it the future we are thinking about turning widgets into fragments so maybe we can do that between.

**Comment by SaumiaSinghal** [comment]
I totally agree @grzesiek2010. If we are having the refreshments of all the widgets at the same time, then we can have those refreshments in sync with the new design proposed for all the widgets. 
I feel once the reworking of the widgets is done, we can then focus on deciding the button and text view layout for the widgets, and then design layout of complete widgets. So, we will be moving step by step while re-designing the widgets right?
